src/AlgAss/AlgMat.jl

```diff
@@ -12,9 +12,9 @@ base_ring(A::MatAlgebra{T, S}) where {T, S} = A.base_ring::parent_type(T)
 
 base_ring_type(::Type{MatAlgebra{T, S}}) where {T, S} = parent_type(T)
 
-coefficient_ring(A::MatAlgebra) = A.coefficient_ring
+coefficient_ring(A::MatAlgebra{T, <:MatrixElem{S}}) where {T, S} = A.coefficient_ring::base_ring_type(S)
```

Comment:
```suggestion
coefficient_ring(A::MatAlgebra{T, <:MatrixElem{S}}) where {T, S} = A.coefficient_ring::parent_type(S)
```
Or alternatively take the base_ring_type of the second type param of MatAlgebra

Reply:
Ah, well spotted, that was me revising my patch in completely: this code used to read
```suggestion
coefficient_ring(A::MatAlgebra{T, S}) where {T, S} = A.coefficient_ring::base_ring_type(S)
```
which however requires a newer AA release adding that `base_ring_type`. So I meant to rewrite it, and apparently didn't finish (I was working on this stuff on an off for over a week, always half an hour here or there, always being interrupted, and last night I just decided to just submit it, no matter what. There is plenty more to be done, but I am not sure when I'll get to it)

Reply:
I prepared a new AA release in https://github.com/Nemocas/AbstractAlgebra.jl/pull/1708 to make the fix available